sycl/source/builtins_generator.py

```diff
@@ -1277,7 +1281,8 @@ def get_template_args(arg_types):
 
 def get_deprecation(builtin, return_type, arg_types):
   """Gets the deprecation statement for a given builtin."""
-  # TODO: Check builtin for deprecation message and prioritize that.
+  if hasattr(builtin, 'deprecation_message') and builtin.deprecation_message:
```

Comment:
Another option to consider, just for future usability, would be to add the `deprecation_message` member to `DefCommon` instead. In that case we don't have to check `hasattr` for `builtin` as all builtins would have the member, even if only `Def` currently sets it.